### PR TITLE
CSM-5174: Code change to exclude the projects in gsuite folder

### PR DESCRIPTION
**Pull request by llukalapu-uptycs**
* This change excludes the projects to integrate with uptycs in the path `Organization root > system-gsuite > apps-script`

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -6,9 +6,26 @@ data "google_project" "my_host_project" {
...
+}
+
+data "google_folders" "app_script_folders" {
+  count = length(local.system_gsuite_folders) == 0 ? 0 : 1
+  parent_id = local.system_gsuite_folders[0]
+}
+
 locals {
+  # Get gsuite folder in organization 
+  system_gsuite_folders = [for each in data.google_folders.my-org-folders.folders : each.name if each.display_name == "uptycs"]
```
Comment (gbonthu) [inline main.tf]: "uptycs"?
Did you check-in your test code? :-)

File: main.tf [hunk 2]
```diff
@@ -6,9 +6,26 @@ data "google_project" "my_host_project" {
...
+  # Get appscript folder in the gsuite folder
+  system_app_script_folders = length(local.system_gsuite_folders) == 0 ? []: [for each in data.google_folders.app_script_folders[0].folders : each.name if each.display_name == "uptycs-sub-folder1"]
```
Comment (gbonthu) [inline main.tf]: test code?

File: versions.tf [hunk 3]
```diff
@@ -20,7 +20,7 @@ terraform {
 
     google = {
       source  = "hashicorp/google"
-      version = "~> 3.61.0"
+      version = ">= 3.61.0"
```
Comment (gbonthu) [inline versions.tf]: why can't we move to the latest?

**Comment by gbonthu** [comment]
```
module "folder_id_reader" {
  source = "./modules/read_folder_id"
  organization_id = var.organization_id
  parent_folder_name = "system-gsuite"
  folder_name = "apps_script"
}
```

can you add a local module in something like `./modules/read_folder_id/...`?
And call that like above? use the output?

File: main.tf (hunk 1 above)
Comment (llukalapu-uptycs) [reply main.tf]: yes

File: versions.tf (hunk 3 above)
Comment (llukalapu-uptycs) [reply versions.tf]: we can use. I will change

File: README.md [hunk 4]
```diff
@@ -1,126 +1,127 @@
 # Terraform GCP module - Organization Integration for Uptycs
 
-## Overview
+This module provides the required GCP resources to integrate a GCP organization with Uptycs.
 
-This module will provision the required GCP resources inorder to integrate GCP organization with Uptycs.
```
Comment (gbonthu) [inline README.md:5]: Keep this please

File: README.md [hunk 5]
```diff
@@ -1,126 +1,127 @@
...
-* You need to have the following privileges to apply the configuration
   * Organization Administrator
   * IAM Workload Identity Pool Admin (at Org level)
   * Service Account Admin (at Org level)
+- You need to Enable following APIs in the host-project:
+  * IAM Service Account Credentials API
+  * Cloud Resource Manager API
+  * Cloud Pub/Sub API
+- `Terraform` version should be >= 1.2.5.
+- `Gcloud` is required for authenticatication and to generate the credentials file.
```
Comment (gbonthu) [inline README.md]:
```suggestion
- `gcloud` is required for authenticatication and to generate the credentials file.
```

File: README.md [hunk 6]
```diff
@@ -1,126 +1,127 @@
...
+Ensure you have the following before you execute the Terraform Script:
 
-### 1. User & IAM
+- You need to have the following privileges to apply the configuration:
 
-* You need to have the following privileges to apply the configuration
   * Organization Administrator
   * IAM Workload Identity Pool Admin (at Org level)
   * Service Account Admin (at Org level)
+- You need to Enable following APIs in the host-project:
```
Comment (gbonthu) [inline README.md]:
```suggestion
- You need to enable the following APIs on the host-project:
```

File: README.md [hunk 7]
```diff
@@ -1,126 +1,127 @@
...
+  * IAM Service Account Credentials API
+  * Cloud Resource Manager API
+  * Cloud Pub/Sub API
```
Comment (gbonthu) [inline README.md]:
```suggestion
  * Cloud Pub/Sub API (Conditional)
```

File: README.md [hunk 8]
```diff
@@ -1,126 +1,127 @@
...
+     organization_id           = "<GCP-ORGANIZATION-ID>"
+     integration_name          = "uptycs-int-20220101"
+     service_account_name      = "sa-for-uptycs"
 
-### 1. Prepare .tf file
+     # Select an existing project from your organization to host resources created by this configuration
+     host_project_id = "<Host-Project-ID>"
 
-* Create a `main.tf` file in a new folder. Paste following configuration and modify as needed.
+     # Set to true, If you want to give permission at organization level
```
Comment (gbonthu) [inline README.md]:
```suggestion
     # Set this to true If you want to give permission at organization level
```

File: README.md [hunk 9]
```diff
@@ -1,126 +1,127 @@
...
+     # Set to false, If you want to give permissions at project level
```
Comment (gbonthu) [inline README.md]:
```suggestion
     # Set this to false otherwise (If you want to give permissions per child project)
```

**Comment by gbonthu** [comment]
Why is `bind_projects_browser_SA_to_Organization ` is configured uncondtionally even when org perms are not set?

File: README.md [hunk 10]
```diff
@@ -1,126 +1,128 @@
...
-#### Outputs
+   ```
 
+**Notes**:
 
-| Name                        | Description                                |
-| ----------------------------- | -------------------------------------------- |
-| credential-file-gen-command | Command to generate credentials JSON file. |
-| host-project-id             | Host Project ID.                           |
+- **If you set the flag `set_org_level_permissions` to `true`, the permissions at the organization level are attached to the service account. Any addition of a project to the organization is automatically integrated with Uptycs. You do not need to execute `step 2`.**
```
Comment (gbonthu) [inline README.md]: This is very confusing.
Are trying to convey that user has to re-run the config if there are new-projects for cases where they did not give us org-permissions?
If so, delete these two lines. Our UI will guide them to re-run TF.
Other option is to add a small note at the end to re-run terraform when new projects are created.
Again having this comment here is very confusing.

**Review by gbonthu — APPROVED** [review]

**Pull request merged**